test/integration/network_reserved_ips_test.go

```diff
@@ -0,0 +1,284 @@
+package integration
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	. "github.com/linode/linodego"
+)
+
+// TestReservedIPAddresses_InsufficientPermissions tests the behavior when a user account
+// doesn't have the can_reserve_ip flag enabled
+func TestReservedIPAddresses_InsufficientPermissions(t *testing.T) {
+	original := os.Getenv("LINODE_TOKEN")
+	dummyToken := "badtoken"
+	os.Setenv("LINODE_TOKEN", dummyToken)
+
+	client, teardown := createTestClient(t, "fixtures/TestReservedIpAddresses_InsufficientPermissions")
+	defer teardown()
+	defer os.Setenv("LINODE_TOKEN", original)
+
+	filter := ""
+	i, getReservedIpsError := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
+	if getReservedIpsError != nil {
+		t.Logf("Error listing ipaddresses, expected struct, got error %v", getReservedIpsError)
+	}
+
+	if len(i) == 0 {
+		t.Logf("Expected a list of ipaddresses, but got none %v", i)
+	}
+
+	// Attempt to reserve an IP address
+	reservedIp, reserveIpError := client.ReserveIPAddress(context.Background(), ReserveIPOptions{
+		Region: "us-east",
+	})
+	if reserveIpError != nil {
+		t.Logf("Failed to reserve IP: %v", reserveIpError)
+	} else {
+		t.Logf("Successfully reserved IP: %+v", reservedIp)
+	}
+
+	// Attempt to get a reserved IP address
+	address := "172.28.3.4"
+	ipaddress, getReservedIpError := client.GetReservedIPAddress(context.Background(), address)
+	if getReservedIpError != nil {
+		t.Logf("Error getting ipaddress, expected struct, got %v and error %v", ipaddress, getReservedIpError)
```

Comment:
Should probably fail the test here.

Reply:
The test now fails if the expected error does not occur.